pkg/kyma/cmd/provision/minikube/hosts_darwin.go

```diff
@@ -11,18 +16,65 @@ const hostsFile = "/etc/hosts"
 
 const defaultVMDriver = vmDriverHyperkit
 
+func isWithSudo() bool {
+	val := os.Getenv("SUDO_UID")
+	if val != "" {
+		return true
+	}
+	return false
+}
+
+func promptUser() bool {
+	for {
+		fmt.Print("Type [y/n]: ")
+		var res string
+		_, err := fmt.Scanf("%s", &res)
+		if err != nil {
+			return false
+		}
+		switch res {
+		case "yes":
+			return true
+		case "no":
+			return false
+		default:
+			continue
+		}
+	}
+}
+
 func addDevDomainsToEtcHostsOSSpecific(o *MinikubeOptions, s step.Step, hostAlias string) error {
-	s.LogInfo("Please enter your password if requested")
+	notifyUserFunc := func() {
+		s.LogInfof("Execute the following command manually to add mappings: sudo sed -i '' \"/"+o.Domain+"/d\" "+hostsFile+" && echo '%s' | sudo tee -a /etc/hosts\r\n", hostAlias)
+		return
+	}
 
+	s.LogInfo("Adding aliases to your 'hosts' file")
+	if isWithSudo() {
+		s.LogInfo("You're running CLI with sudo. CLI has to add Kyma domain entries to your 'hosts'. Type 'y' to allow this action")
+		if !promptUser() {
+			notifyUserFunc()
+			return nil
+		}
+	}
 	_, err := internal.RunCmd("sudo", []string{"/bin/sh", "-c", "sed -i '' \"/" + o.Domain + "/d\" " + hostsFile})
 	if err != nil {
-		return err
+		notifyUserFunc()
+		return nil
```

Comment:
Is it ok to just ignore the error here? Maybe it is not permission related.

Reply:
As discussed with @a-thaler.